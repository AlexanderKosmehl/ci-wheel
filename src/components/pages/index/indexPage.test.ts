import { describe, expect, it } from 'vitest';
import generateIndexPage from './indexPage';

describe('generateIndexPage', () => {
  const page = generateIndexPage();

  it('generates component correctly', () => {
    expect(page).toMatchSnapshot();
  });
});
