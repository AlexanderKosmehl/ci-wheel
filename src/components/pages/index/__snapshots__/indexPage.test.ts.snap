// Vitest Snapshot v1

exports[`generateIndexPage > generates component correctly 1`] = `
<div
  class="indexPage"
>
  <header
    class="header"
  >
    <div
      class="logoTitleWrapper"
    >
      <img
        class="headerLogo"
        src="/src/assets/ci-logo.png"
      />
      <h1
        class="headerTitle"
      >
        Wheel
      </h1>
    </div>
    <div
      class="menuButtonWrapper"
    >
      <button
        class="button sidebarToggle"
      >
        <img
          class="icon"
          src="/src/assets/menu-icon.svg"
        />
      </button>
      <button
        class="button sidebarToggle hidden"
      >
        <img
          class="icon"
          src="/src/assets/arrow-back-icon.svg"
        />
      </button>
    </div>
  </header>
  <div
    class="mainContainer"
    id="spinner-container"
  >
    <div
      class="spinnerWrapper"
    >
      <div
        class="container"
      >
        <span
          class="placeholder"
        >
          Keine Eintr√§ge
        </span>
      </div>
    </div>
    <div
      class="sidebar"
      hidden=""
      id="sidebar"
    >
      <div
        class="container"
      >
        <input
          class="input"
          placeholder="Neuer Eintrag"
          type="text"
        />
        <button
          class="button addButton"
          disabled=""
        >
          <img
            class="icon"
            src="/src/assets/add-icon.svg"
          />
        </button>
      </div>
      <ul
        class="listContainer"
      />
      <div
        class="container"
      >
        <button
          class="footerButton"
        >
          Import
        </button>
        <button
          class="footerButton"
        >
          Archiv
        </button>
      </div>
    </div>
  </div>
  <div />
</div>
`;
